### PR TITLE
Math TOP K function with tests

### PR DESCRIPTION
**Pull request by MissingNO57**

### CODE REVIEW

File: libs/math/include/math/top_k.hpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+ * @param data input
+ * @param k number of k-highest values
+ * @param sorted TRUE=descending order, FALSE=ascending order
+ */
+template <typename ArrayDataType, typename ArrayIndicesType>
+void TopK(ArrayDataType &ret_data, ArrayIndicesType &ret_indices, ArrayDataType const &data,
+          typename ArrayDataType::SizeType k, bool sorted = true)
+{
+  using DataType   = typename ArrayDataType::Type;
+  using IndiceType = typename ArrayIndicesType::Type;
```
Comment (baykaner) [inline libs/math/include/math/top_k.hpp]: by the way, the singular of 'indices' is 'index'

File: libs/math/include/math/top_k.hpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  // Set itaration order
+  data_it.MoveAxisToFront(axis);
+  ret_data_it.MoveAxisToFront(axis);
+  ret_indices_it.MoveAxisToFront(axis);
+
+  // Prepare vector for sorting values
+  std::vector<std::pair<SizeType, DataType>> vec;
+  vec.resize(axis_size);
+
+  SizeType i = 0, k_minus_one = k - 1;
```
Comment (baykaner) [inline libs/math/include/math/top_k.hpp]: style wise I think we're suppose to put this kind of statement on two lines for clarity:
SizeType i = 0;
SizeType k_mins_one = k - 1;

File: libs/math/tests/unit/gtest/top_k.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+#include "gtest/gtest.h"
+
+using namespace fetch::math;
+
+template <typename T>
+class TopKTest : public ::testing::Test
+{
+};
+
+using MyTypes = ::testing::Types<fetch::math::Tensor<float>, fetch::math::Tensor<double>,
```
Comment (baykaner) [inline libs/math/tests/unit/gtest/top_k.cpp]: once #1882 goes in this file will need a few slight alterations:
- include test_types & don't include tensor
- namespace everything inside fetch::math::test
- remove MyTypes and use FloatingTypes
this makes it easier for us to test all the floating types we care about without missing any out from each list of types in each test file (and avoids weird bugs from not being properly namespaced)

**Review by baykaner — CHANGES_REQUESTED** [review]

File: libs/math/include/math/top_k.hpp (hunk 1 above)
Comment (MissingNO57) [reply libs/math/include/math/top_k.hpp]: Of course, it is...

File: libs/math/tests/unit/gtest/top_k.cpp (hunk 3 above)
Comment (MissingNO57) [reply libs/math/tests/unit/gtest/top_k.cpp]: I did required changes and waiting for your PR to be merged.

**Review by baykaner — APPROVED** [review]
looks great, nice!

**Pull request merged**